docs/src/main/sphinx/sql/select.rst

```diff
@@ -668,11 +668,18 @@ A window specification has the following components:
   rows from the start of the partition up to the last peer of the current row.
   In the absence of ``ORDER BY``, all rows are considered peers, so ``RANGE
   BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW`` is equivalent to ``BETWEEN
-  UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING``
-
-Each component is optional. In the case when window partitioning, ordering or frame
-is not specified and existing window name is absent or the referenced window
-specification does not provide particular components, defaults are used.
+  UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING``. The window frame syntax
+  supports additional clauses for row pattern recognition. If the row pattern
+  recognition clauses are specified, the window frame for a particular row
+  consists of the rows matched by a pattern starting from that row.
+  Additionally, if the frame specifies row pattern measures, they can be
+  called over the window, similarly to window functions. For more details, see
+  :doc:`Row pattern recognition in window structures
+  </sql/pattern-recognition-in-window>`.
+
+Each window component is optional. Defaults are used when window partitioning,
```

Comment:
The sentence starting at "Defaults ..." is hard to parse. Does the "and" associate more strongly than the last "or". Maybe split up the sentence:

"Defaults are used when the existing window name is absent and the window partitioning, ordering or frame are not specified. They are also used when ..."

Or make them a bullet list:

```
Defaults are used in any of these scenarios:
* ....
* ....
```

By the way, what does "existing window name is absent" mean?

Reply:
"existing window name is absent" refers to the `existing window name` being part of window specification.

I tried to reword this paragraph, but might need more help with it.